pandas/tests/indexes/test_setops.py

```diff
@@ -50,13 +50,11 @@ def test_union_different_types(index_flat, index_flat2, request):
     if (
         not idx1.is_unique
         and not idx2.is_unique
-        and not idx2.is_monotonic_decreasing
```

Comment:
could add a comment pointing back here to the effect of "`not idx2.is_monotonic_decreasing` covers _most_ cases but is too strict"